components/cloud_start_trial/cloud_start_trial_btn.tsx

```diff
@@ -91,11 +88,19 @@ const CloudStartTrialButton = ({
             return TrialLoadStatus.Failed;
         }
 
-        await dispatch(getCloudSubscription());
-        await dispatch(getClientConfig());
+        function ensureUpdatedData() {
+            // Depending on timing of pods rolling, the webhook may still not get sent.
+            // Re-request limits as a just-in-case, but only well after any
+            // pods still alive should have either purged cache,
+            // updated limits, or be brand new pods that won't be holding onto stale limits
+            // We don't need to re-request subscription: the updated value is sent in the
+            // request cloud trial response.
+            // We don't need to request license: its update process is independent
+            // from subscription/limit changes and always happens after pods roll.
+            dispatch(getCloudLimits());
+        }
 
-        await dispatch(getLicenseConfig());
-        await dispatch(getCloudLimits());
+        setTimeout(ensureUpdatedData, 5000);
```

Comment:
I wanted it to be long i.e. at least 5 seconds because I was seeing that immediate re-request could lead to getting the stale starter limits if the node that was contacted was the node that didn't handle the request. You can see an example of that in the picture of the logs attached to the enterprise PR.

Reply:
I could extract the 5000 ms as a const named something like `TIME_UNTIL_CACHE_PURGE_GUESS`

Reply:
that would help for readability definitely. thanks!

Reply:
Are these 5 seconds in addition to the current wait time we had or is this in replacement of the time we waiting to request license and subscription data?

Reply:
I don't think we had a well defined wait time before. Instead, we had 4 requests (subscription, license, config, limits, and limits came last), and each waited for the other to complete. So these 5 seconds are a replacement of that. But another difference is that this is wrapped in a setTimeout, so this is asynchronous where the other requests happened synchronously because of the `await`s.